src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -978,36 +978,23 @@ void MacroAssembler::li(Register Rd, int64_t imm) {
   }
 }
 
+void MacroAssembler::load_link_jump(const address source, Register temp) {
+  assert(temp != noreg && temp != x0, "expecting a register");
+  assert_cond(source != nullptr);
+  int64_t distance = source - pc();
+  assert(is_simm32(distance), "Must be");
+  Assembler::auipc(temp, (int32_t)distance + 0x800);
+  Assembler::_ld(temp, temp, ((int32_t)distance << 20) >> 20);
```

Comment:
Question: Why would you use this low-level `Assembler::_ld` here?

Reply:
I used to be excplicit about this is the normal **ld**.
But I see I did not do the same for **jalr** => **_jalr**. (as we are in MASM we can use 'private' method).
I'll change to normal ld and add an assert that we are in a incompressable region?

(I would like to revert that at some time, so the user of reloc_call don't need to know about it needs incompressable for reloc_call)

Suggested:
```
@@ -982,0 +983 @@ void MacroAssembler::load_link_jump(const address source, Register temp) {
+  assert(!in_compressible_region(), "Must be");
@@ -987 +988 @@ void MacroAssembler::load_link_jump(const address source, Register temp) {
-  Assembler::_ld(temp, temp, ((int32_t)distance << 20) >> 20);
+  Assembler::ld(temp, temp, ((int32_t)distance << 20) >> 20);
```

Reply:
Seems no need to worry about that? I think it's up to the user to decide if it wants compressed instructions. Here in this case, the only call site is within a relocate() which will disable compressed instructions for you[1].
```
  relocate(entry.rspec(), [&] {
    load_link_jump(target);
  });
```
[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/assembler_riscv.hpp#L2130

Reply:
When you use reloc_call the size of the call must be an exact size we already specified (3 * NativeInstruction::instruction_size). (if those sizes don't apply you don't want a reloc_call) 
So there is nothing to choose from the caller prespective.

Reply:
Maybe in the future load_link_jump will be used out side reloc_call, was that your thinking?

Anyhow change _ld to ld ?

Reply:
Yeah, I just feel it's better to decouple the two and let the call site to decide.

Reply:
I just changed to ld.